### ISSUE TITLE
tr: different behavior than GNU with `echo "BB%t"|tr -c -d -s Bt "B" `

### ISSUE THREAD

**Issue by sylvestre** [uutils/coreutils#6163]
Playing with a pending tr fuzzer, it found this issue:
With GNU impl:
```
$ echo "BB%t"|/usr/bin/tr -c -d -s Bt "B"
Bt%
```

With our:
```
$ echo "BB%t"|tr -c -d -s Bt "B"
BBt%                                                 
```

**Comment by BenWiederhake** [comment]
Is it possible that there are special unicode-characters or copy-paste errors involved? I get completely different results:

```console
$ echo "BB%t" | cargo run tr -c -d -s Bt "B"
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/coreutils tr -c -d -s Bt B`
Bt
$ echo "BB%t" | tr -c -d -s Bt "B"
Bt
```

The percent character seems to move from *before* the `t` to *after* the `t` in both of your calls. D:

**Comment by sylvestre** [comment]
hmm,let me try to find a different test case with https://github.com/uutils/coreutils/pull/6167

**Comment by sylvestre** [comment]
Another example:
with GNU:
```
$ echo "1ÃoZrkbJhoW1x(ebH0(57mOX500(wLIImI((y44I9yOT2bioGzdasv6UZFjqQ(Uha3VF0yx4nDO4EtÃAigGe(vzs(SoeHetn(G"|/usr/bin/tr -c -s -t k3yC SIU
1ÃoZrkbJhoW1x(ebH0(57mOX500(wLImI((y44I9yOT2bioGzdasv6UZFjqQ(Uha3VF0yx4nDO4EtÃAigGe(vzs(SoeHetn(G

```

With Rust:
```
$ echo "1ÃoZrkbJhoW1x(ebH0(57mOX500(wLIImI((y44I9yOT2bioGzdasv6UZFjqQ(Uha3VF0yx4nDO4EtÃAigGe(vzs(SoeHetn(G"|./target/debug/coreutils tr -c -s -t k3yC SIU
UkUyUyU3UyU%                                             
```

**Comment by BenWiederhake** [comment]
Yes, this time I can reproduce it and minimize it:
```console
$ echo X | tr -c -t Y Z
X
$ echo X | cargo run tr -c -t Y Z
ZZ
```

Initially I suspected that it's one of the many bugs in #6133, but looks like this really is a completely new bug xD

**Comment by jalil-salame** [comment]
I fixed this in #6340 my guess is that GNU tr calculates the complement of set1/ARRAY1 and truncates that (so `tr -c -t Y Z` is equivalent to `tr '\0' Z`). I verified this with `print '\0' | tr -c -t Y Z`.
